libcomcat/classes.py

```diff
@@ -674,11 +675,13 @@ def toDict(
                 edict.update(_get_focal_mechanism_info(focal))
 
         if get_all_magnitudes:
+            phases = []
             if self.hasProduct("phase-data"):
-                product = "phase-data"
-            else:
-                product = "origin"
-            phases = self.getProducts(product, source="all")
+                phase_products = self.getProducts("phase-data", source="all")
+                phases += phase_products
+            if self.hasProduct("origin"):
+                origin_products = self.getProducts("origin", source="all")
+                phases += origin_products
```

Comment:
If `phases` has both phase products and origin products (some of which might hold the same information), how do we ensure we aren't getting duplicate data? Is there ever a case where the origin would have the same source and mag type but a different value from the phase data? Is there ever a case where the user would want to specify which product type (origin or phase data) is used to get magnitude information?

Reply:
origin and phase-data products with the same source+code+updateTime were both generated from the phase-data product.  contributors are more likely to send origin without phase-data than the opposite.

On event pages we start with the list of origins, then get details from the more complete phase-data when available:
https://code.usgs.gov/ghsc/hazdev/earthquake-eventpages/-/blob/master/src/app/event.ts#L29